pandas/tests/test_window.py

```diff
@@ -289,6 +290,233 @@ def test_deprecations(self):
+            data['df'] = df
+
+        return data
+
+    def _create_data(self):
+        super(TestDtype, self)._create_data()
+        self.data = self._create_dtype_data(self.dtype)
+        self.expects = self.get_expects()
+
+    def setUp(self):
+        self._create_data()
+
+    def _cast_result(self, result, from_dtype, to_dtype):
+        if com.needs_i8_conversion(from_dtype):
+            if isinstance(result, Series):
+                result = result.view('i8')
+            elif isinstance(result, DataFrame):
+                final = []
+                for idx in result:
+                    final.append(Series(result[idx].view('i8')))
+                result = pd.concat(final, axis=1).reindex(
+                    columns=result.columns)
+        return result.astype(to_dtype)
+
+    def test_dtypes(self):
+        for f_name, d_name in product(self.funcs.keys(), self.data.keys()):
+            # Specify if the results and expectations
+            # need to be coerced to a given dtype
+            # once we changed the return value for roll_<function>,
+            # we should change coerce behavior here accordingly
+            f, res_dtype, exp_dtype = self.funcs[f_name]
+            d = self.data[d_name]
+            assert_equal = assert_series_equal if isinstance(
+                d, Series) else assert_frame_equal
+            exp = self.expects[d_name][f_name]
+            if exp_dtype:
+                exp = exp.astype(com.pandas_dtype(exp_dtype))
+
+            roll = d.rolling(window=self.window)
+            result = f(roll)
+            if res_dtype:
+                result = self._cast_result(result,
+                                           self.dtype,
+                                           com.pandas_dtype(res_dtype))
+            assert_equal(result, exp)
+
+
+class TestDtype_object(TestDtype):
+    dtype = object
+
+
+class TestDtype_int8(TestDtype):
+    dtype = np.int8
+
+
+class TestDtype_int16(TestDtype):
+    dtype = np.int16
+
+
+class TestDtype_int32(TestDtype):
+    dtype = np.int32
+
+
+class TestDtype_int64(TestDtype):
+    dtype = np.int64
+
+
+class TestDtype_uint8(TestDtype):
+    dtype = np.uint8
+
+
+class TestDtype_uint16(TestDtype):
+    dtype = np.uint16
+
+
+class TestDtype_uint32(TestDtype):
+    dtype = np.uint32
+
+
+class TestDtype_uint64(TestDtype):
+    dtype = np.uint64
+
+
+class TestDtype_float16(TestDtype):
+    dtype = np.float16
+
+
+class TestDtype_float32(TestDtype):
+    dtype = np.float32
+
+
+class TestDtype_float64(TestDtype):
+    dtype = np.float64
+
+
+class TestDtype_timedelta(TestDtype):
+    dtype = np.dtype('m8[ns]')
+    import pandas.tslib as tslib
+    naval = tslib.iNaT
+
+    funcs = {
+        'count': (lambda v: v.count(), None, 'float64'),
+        'max': (lambda v: v.max(), None, 'm8[ns]'),
+        'min': (lambda v: v.min(), None, 'm8[ns]'),
+        'sum': (lambda v: v.sum(), None, 'm8[ns]'),
```

Comment:
so what you need to do is make a sub-class of ``TestDtype`` that is ``TestDatetimeLikeDtype`` (and put ``m8/M8/M8 w/tz``) underneath